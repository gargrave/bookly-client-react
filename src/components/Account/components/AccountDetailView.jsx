import React from 'react'
import { shape, string } from 'prop-types'

const AccountDetailView = (props) => {
  const { email } = props.user
  return (
    <div className="account-detail-view">
      <h2>AccountDetailView</h2>
      <p className="email-display">
        <strong>Email: </strong>
        {email}
      </p>
    </div>
  )
}

AccountDetailView.propTypes = {
  user: shape({
    email: string.isRequired,
  }).isRequired,
}

export default AccountDetailView
