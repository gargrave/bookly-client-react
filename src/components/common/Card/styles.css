.bookly-card {
  background-color: #f9f9f9;
  cursor: pointer;
  border: solid 1px #bbb;
  border-radius: 5px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  margin: 10px auto;
  width: 95%;
  max-width: 600px;
  padding: 8px;
}

.bookly-card:hover {
  background: #ffe8e4;
}

.bookly-card p {
  margin: 0;
}

.bookly-card .bookly-card-title {
  font-weight: bold;
}

.bookly-card .bookly-card-text {
  color: #6e6969;
}
