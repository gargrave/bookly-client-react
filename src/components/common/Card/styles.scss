.bookly-card {
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  box-shadow:
    0 1px 3px rgba(0, 0, 0, 0.07),
    0 1px 2px rgba(0, 0, 0, 0.12);
  margin: 8px 10px;
  padding: 10px 20px;
  width: 95%;

  hr {
    margin: 12px 0;
  }

  &.bookly-hoverable:hover {
    background: #ffe8e4;
    cursor: pointer;
  }

  p {
    margin: 0;
  }

  &-title {
    font-weight: bold;
  }

  &-text {
    color: #6e6969;
  }

  &-text + &-text {
    margin-top: 8px;
  }

  &-spacer {
    height: 20px;
    line-height: 20px;
  }

  &.bookly-detail-card {
    .bookly-card-title {
      font-size: 1.3rem;
      line-height: 2.5rem;
    }

    .bookly-card-title + .bookly-card-text {
      font-size: 1.1rem;
    }
  }
}
