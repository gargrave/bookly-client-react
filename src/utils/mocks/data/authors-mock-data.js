export const authorsMockData = [
  {
    id: 8,
    firstName: 'Aldous',
    lastName: 'Huxley',
    createdAt: '2017-07-01T01:16:43.881Z',
    updatedAt: '2017-07-01T01:16:43.881Z'
  },
  {
    id: 51,
    firstName: 'Kurt',
    lastName: 'Vonnegut',
    createdAt: '2017-07-06T22:02:11.504Z',
    updatedAt: '2017-07-06T22:02:11.504Z'
  },
  {
    id: 59,
    firstName: 'George R. R.',
    lastName: 'Martin',
    createdAt: '2017-08-17T20:47:35.232Z',
    updatedAt: '2017-08-17T20:47:35.232Z'
  },
  {
    id: 52,
    firstName: 'Tom',
    lastName: 'Robbins',
    createdAt: '2017-07-06T22:02:17.051Z',
    updatedAt: '2017-08-17T21:27:35.885Z'
  }
]
