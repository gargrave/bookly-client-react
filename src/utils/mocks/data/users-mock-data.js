export const usersMockData = [
  {
    id: 1,
    email: 'mockuser1@gmail.com',
    createdAt: '2017-06-30T19:52:54.644Z',
    updatedAt: '2017-06-30T19:52:54.644Z',
    lastLogin: '2017-07-01T01:11:03.942Z',
    verified: true,
  },
];
