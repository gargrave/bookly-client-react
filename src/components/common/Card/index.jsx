import React from 'react'
import { array, func, string } from 'prop-types'

import { buildClasses } from '@/utils/cssHelpers'

import './styles.css'

const renderText = (text, classname) => (
  text
    ? <p className={buildClasses(classname)}>{text}</p>
    : null
)

const Card = ({
  children,
  classes = [],
  onClick,
  text,
  title,
}) => {
  return (
    <div
      className={buildClasses(['card', ...classes])}
      onClick={onClick}>
      {renderText(title, 'card-title')}
      {renderText(text, 'card-text')}
      {children}
    </div>
  )
}

Card.propTypes = {
  children: array,
  classes: array,
  onClick: func.isRequired,
  text: string,
  title: string,
}

export default Card
