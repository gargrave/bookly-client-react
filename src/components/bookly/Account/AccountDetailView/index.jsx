import React from 'react'
import { shape, string } from 'prop-types'
import { format } from 'date-fns'

import { buildClasses } from '@/utils/cssHelpers'

import Card from '@/components/common/Card'

const AccountDetailView = ({ user }) => {
  return (
    <div className={buildClasses('account-detail-view')}>
      <Card
        classes={['detail-card', 'account-detail-card']}
        hoverable={false}
        onClick={() => null}
        title="My Account">
        <p className={buildClasses('card-text')}>
          <strong>Email:</strong> {user.email}
        </p>
        <hr/>
        <p className={buildClasses('card-text')}>
          <strong>Registered:</strong> {format(user.createdAt, 'MMM. DD, YYYY, HH:mm:ss')}
        </p>
        <p className={buildClasses('card-text')}>
          <strong>Updated:</strong> {format(user.updatedAt, 'MMM. DD, YYYY, HH:mm:ss')}
        </p>
      </Card>
    </div>
  )
}

AccountDetailView.propTypes = {
  user: shape({
    email: string.isRequired,
    createdAt: string,
    updatedAtmail: string,
  }).isRequired,
}

export default AccountDetailView
