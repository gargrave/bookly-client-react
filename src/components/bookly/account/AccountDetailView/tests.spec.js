import React from 'react'
import { shallow } from 'enzyme'

import { usersMockData } from '@/utils/mocks/data/users-mock-data'

import Card from '@/components/common/Card'

import AccountDetailView from './'

describe('AccountDetailView', () => {
  let props
  let component

  beforeEach(() => {
    props = {
      onLogoutClick: jest.fn(),
      user: Object.create(usersMockData[0]),
    }

    component = shallow(<AccountDetailView {...props} />)
  })

  test('renders correctly', () => {
    expect(component).toMatchSnapshot()
    expect(component.find('.bookly-account-detail-view').length).toEqual(1)
    expect(component.find(Card).length).toEqual(1)
  })
})
