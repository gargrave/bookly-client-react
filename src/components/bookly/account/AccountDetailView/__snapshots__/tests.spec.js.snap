// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetailView renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountDetailView
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        classes={
          Array [
            "detail-card",
            "account-detail-card",
          ]
        }
        hoverable={false}
        onClick={[Function]}
        title="My Account"
      >
        <p
          className="bookly-card-text"
        >
          <strong>
            Email:
          </strong>
           
          mockuser1@gmail.com
        </p>
        <hr />
        <p
          className="bookly-card-text"
        >
          <strong>
            Registered:
          </strong>
           
          Jun. 30, 2017, 12:52:54
        </p>
        <p
          className="bookly-card-text"
        >
          <strong>
            Updated:
          </strong>
           
          Jun. 30, 2017, 12:52:54
        </p>
      </Card>,
      "className": "bookly-account-detail-view",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <p
            className="bookly-card-text"
          >
            <strong>
              Email:
            </strong>
             
            mockuser1@gmail.com
          </p>,
          <hr />,
          <p
            className="bookly-card-text"
          >
            <strong>
              Registered:
            </strong>
             
            Jun. 30, 2017, 12:52:54
          </p>,
          <p
            className="bookly-card-text"
          >
            <strong>
              Updated:
            </strong>
             
            Jun. 30, 2017, 12:52:54
          </p>,
        ],
        "classes": Array [
          "detail-card",
          "account-detail-card",
        ],
        "hoverable": false,
        "onClick": [Function],
        "title": "My Account",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Email:
              </strong>,
              " ",
              "mockuser1@gmail.com",
            ],
            "className": "bookly-card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email:",
              },
              "ref": null,
              "rendered": "Email:",
              "type": "strong",
            },
            " ",
            "mockuser1@gmail.com",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Registered:
              </strong>,
              " ",
              "Jun. 30, 2017, 12:52:54",
            ],
            "className": "bookly-card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Registered:",
              },
              "ref": null,
              "rendered": "Registered:",
              "type": "strong",
            },
            " ",
            "Jun. 30, 2017, 12:52:54",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Updated:
              </strong>,
              " ",
              "Jun. 30, 2017, 12:52:54",
            ],
            "className": "bookly-card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Updated:",
              },
              "ref": null,
              "rendered": "Updated:",
              "type": "strong",
            },
            " ",
            "Jun. 30, 2017, 12:52:54",
          ],
          "type": "p",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          classes={
            Array [
              "detail-card",
              "account-detail-card",
            ]
          }
          hoverable={false}
          onClick={[Function]}
          title="My Account"
        >
          <p
            className="bookly-card-text"
          >
            <strong>
              Email:
            </strong>
             
            mockuser1@gmail.com
          </p>
          <hr />
          <p
            className="bookly-card-text"
          >
            <strong>
              Registered:
            </strong>
             
            Jun. 30, 2017, 12:52:54
          </p>
          <p
            className="bookly-card-text"
          >
            <strong>
              Updated:
            </strong>
             
            Jun. 30, 2017, 12:52:54
          </p>
        </Card>,
        "className": "bookly-account-detail-view",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <p
              className="bookly-card-text"
            >
              <strong>
                Email:
              </strong>
               
              mockuser1@gmail.com
            </p>,
            <hr />,
            <p
              className="bookly-card-text"
            >
              <strong>
                Registered:
              </strong>
               
              Jun. 30, 2017, 12:52:54
            </p>,
            <p
              className="bookly-card-text"
            >
              <strong>
                Updated:
              </strong>
               
              Jun. 30, 2017, 12:52:54
            </p>,
          ],
          "classes": Array [
            "detail-card",
            "account-detail-card",
          ],
          "hoverable": false,
          "onClick": [Function],
          "title": "My Account",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Email:
                </strong>,
                " ",
                "mockuser1@gmail.com",
              ],
              "className": "bookly-card-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email:",
                },
                "ref": null,
                "rendered": "Email:",
                "type": "strong",
              },
              " ",
              "mockuser1@gmail.com",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Registered:
                </strong>,
                " ",
                "Jun. 30, 2017, 12:52:54",
              ],
              "className": "bookly-card-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Registered:",
                },
                "ref": null,
                "rendered": "Registered:",
                "type": "strong",
              },
              " ",
              "Jun. 30, 2017, 12:52:54",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Updated:
                </strong>,
                " ",
                "Jun. 30, 2017, 12:52:54",
              ],
              "className": "bookly-card-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Updated:",
                },
                "ref": null,
                "rendered": "Updated:",
                "type": "strong",
              },
              " ",
              "Jun. 30, 2017, 12:52:54",
            ],
            "type": "p",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
